plugin/src/main/java/org/opensearch/ml/rest/AbstractMLSearchAction.java

```diff
@@ -46,12 +58,17 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
         searchSourceBuilder.parseXContent(request.contentOrSourceParamParser());
         searchSourceBuilder.fetchSource(getSourceContext(request, searchSourceBuilder));
         searchSourceBuilder.seqNoAndPrimaryTerm(true).version(true);
+        if (mlFeatureEnabledSetting.isMultiTenancyEnabled()) {
+            // Add tenant ID filter
+            String tenantId = getTenantID(mlFeatureEnabledSetting.isMultiTenancyEnabled(), request);
+            searchSourceBuilder.query(QueryBuilders.termQuery(TENANT_ID, tenantId));
+        }
         SearchRequest searchRequest = new SearchRequest().source(searchSourceBuilder).indices(index);
         return channel -> client.execute(actionType, searchRequest, search(channel));
```

Comment:
I'm really not fond of putting logic in the Rest Handlers, as the `MachineLearningClient` doesn't call them; it directly invokes the relevant transport action.

In particular, while this works for the ML Commons REST API, we won't be able to search for an agent from Flow Framework or any other external plugin that consumes the MLClient; we'd essentially have to duplicate all this code on our side to use the search API.

Since Rest Handlers don't benefit from injection, this also means passing more parameters around in constructors.

We're already a ways down this road with all the other logic building the query here, so for this PR I guess it's OK, but this really should be refactored into a transport action where the feature enabled settings can be injected and it can be called from a client.

Reply:
that's a good point. In the transport layer, we can check again, if `mlFeatureEnabledSetting.isMultiTenancyEnabled()` is true, we should expect tenant filtering in the search request. Otherwise throw exception?

Reply:
Yes.  Problem is such an action doesn’t yet exist.   I’m ok with this as is with an open issue to fix it long term.

Reply:
I added the check in the transport search layer. Could you please review again?

Reply:
Rather than adding the term query check in rest/transport actions, can it be added as part of sdkclient? this will prevent code bugs by ensuring new code flows does not miss out on adding the check by implementing the term condition at the lowest layer.